lib/ws/ws.js

```diff
@@ -9,13 +9,86 @@ var factory = require('./factory');
 var authMethod = 'auth';
 
 /**
+ * race takes an array of promises and is resolved with the first promise
+ * that resolves, or rejects. After the first resolve or rejection, all
+ * future resolutions and rejections will be discarded.
+ * @param {Promise[]} promises
+ * @return {Promise}
+ */
+function race(promises) {
```

Comment:
Generally not a fan of using different libraries that do the same thing across multiple projects, it would also have prevented you having to re-inventing those functions.

However, if you really want to keep it agnostic, you may want to allow the end user of the library to inject their own promise implementation maybe via `require('beam-client-node').use('Promise', PROMISE_CONSTRUCTOR);`, this way you can drop the bluebird dependency.